Update several python versions in recipe/meta.yaml

yantosca opened the pull request:
Updated several Python package versions
- numpy           (1.2.6 or greater)
- cartopy         (1.26.4 or greater)
- dask            (2024.5.2 or greater)
- esmf/esmpy      (8.6.1 or greater)
- gridspec        (0.1.0)
- joblib          (1.4.2 or greater)
- matplotlib      (3.8.4 or greater)
- matplotlib-base (3 or greater)
- pandas          (2.2.2 or greater)
- scipy           (1.13.1 or greater)
- sparselt        (0.1.3 or greater)
- tabulate        (0.9.0 or greater)
- tk              (8.6.13 or greater)
- xarray          (2024.5.0 or greater)
- xesmf           (0.8.5 or greater)

Also note: pypdf2 is now replaced by pypdf.  This was causing builds to fail.

<!--
@conda-forge-admin, please rerender
-->

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
This

yantosca committed:
Update several python versions in recipe/meta.yaml

Updated
- numpy           (1.2.6 or greater)
- cartopy         (1.26.4 or greater)
- dask            (2024.5.2 or greater)
- esmf/esmpy      (8.6.1 or greater)
- gridspec        (0.1.0)
- joblib          (1.4.2 or greater)
- matplotlib      (3.8.4 or greater)
- matplotlib-base (3 or greater)
- pandas          (2.2.2 or greater)
- scipy           (1.13.1 or greater)
- sparselt        (0.1.3 or greater)
- tabulate        (0.9.0 or greater)
- tk              (8.6.13 or greater)
- xarray          (2024.5.0 or greater)
- xesmf           (0.8.5 or greater)

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Bumped build number to 3 in meta.yaml

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ ``requirements: run: gridspec==0.1.0`` must contain a space between the name and the pin, i.e. ``gridspec ==0.1.0``

For **recipe/meta.yaml**:

 * ℹ️ Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.
 * ℹ️ No valid build backend found for Python recipe for package `geoschem-gcpy` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.
 * ℹ️ top-level output has some malformed specs:
- In section run: `gridspec==0.1.0, `matplotlib-base >=3.
Requirements spec fields should always be space-separated to avoid known issues in conda-build. For example, instead of `name =version=build`, use `name version build`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639767851](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639767851). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you but ran into some issues. Please check the [output logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639716986) of the GitHub Actions workflow below for more details. You can also ping `conda-forge/core` (using the `@` notation) for further assistance or you can try [rerendering locally](https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-with-conda-smithy-locally).


The following suggestions might help debug any issues:
* Is the `recipe/{meta.yaml,recipe.yaml}` file valid?
* If there is a `recipe/conda-build-config.yaml` file in the feedstock make sure that it is compatible with the current [global pinnnings](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
* Is the fork used for this PR on an organization or user GitHub account? Automated rerendering via the webservices admin bot only works for user GitHub accounts.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639716986](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639716986). Examine the logs at this URL for more detail.</sub>

yantosca committed:
Bug fix: ">==" -> ">=" in dask specification

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!

yantosca committed:
Add a space after gridspec

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Removed matplotlib

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ top-level output has some malformed specs:
- In section run: `matplotlib-base >=3.
Requirements spec fields should always be space-separated to avoid known issues in conda-build. For example, instead of `name =version=build`, use `name version build`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639924827](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13639924827). Examine the logs at this URL for more detail.</sub>

yantosca committed:
Further edits to meta.yaml

- Add MPLBACKEND=agg to the build section
- Add setuptools to the host section

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Bug fix: Now use matplotlib-base >= 3.8.4

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

I considered the following status checks when analyzing this PR:
 - **linter**: passed
 - **azure**: passed


Thus the PR was passing and merged! Have a great day!

yantosca committed:
Strip version numbers

- Bump build number to 5
- Strip version numbers to avoid Python conflicts

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!

yantosca committed:
Add requests to the meta.yaml

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Try xesmf >= 0.8.0 to avoid build error w/ Python 13

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ ``requirements: run: xesmf >= 0.8.0`` should not contain a space between relational operator and the version, i.e. ``xesmf >=0.8.0``

For **recipe/meta.yaml**:

 * ℹ️ top-level output has some malformed specs:
- In section run: `xesmf >= 0.8.0
Requirements spec fields should always be space-separated to avoid known issues in conda-build. For example, instead of `name =version=build`, use `name version build`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13640575854](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13640575854). Examine the logs at this URL for more detail.</sub>

yantosca committed:
"xesmf >= 0.8.0" --> "xesmf >= 0.8.0" in meta.yaml

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

yantosca committed:
Try pinning esmpy <8.4 to avoid "import ESMF" error

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Suspend builds with Python 3.13 for now

.azure.pipelines/*.yml
- Commented out the commands to build with Python 13

recipe/meta.yml
- Unpin all packages

README.md
- Updated package maintainers

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

Commits were made to this PR after the `automerge` label was added. For security reasons, I have disabled automerge by removing the `automerge` label. Please add the `automerge` label again (or ask a maintainer to do so) if you'd like to enable automerge again!
